app/src/ui/preferences/appearance.tsx

```diff
@@ -50,7 +54,45 @@ const themes: ReadonlyArray<ISegmentedItem<ApplicationTheme>> = [
   ...(supportsSystemThemeChanges() ? [systemTheme] : []),
 ]
 
-export class Appearance extends React.Component<IAppearanceProps, {}> {
+export class Appearance extends React.Component<
+  IAppearanceProps,
+  IAppearanceState
+> {
+  public constructor(props: IAppearanceProps) {
+    super(props)
+
+    const usePropTheme =
+      props.selectedTheme !== ApplicationTheme.System ||
+      supportsSystemThemeChanges()
+
+    this.state = { selectedTheme: usePropTheme ? props.selectedTheme : null }
+
+    if (!usePropTheme) {
+      this.initializeSelectedTheme()
+    }
+  }
+
+  public async componentDidUpdate(prevProps: IAppearanceProps) {
+    if (prevProps.selectedTheme === this.props.selectedTheme) {
+      return
+    }
+
+    const usePropTheme =
+      this.props.selectedTheme !== ApplicationTheme.System ||
+      supportsSystemThemeChanges()
```

Comment:
I recognize that this is the exact same logic from before, only in its inversed form but I can't understand it. Why do we want to `usePropTheme` when `supportsSystemThemeChanges` is true?

Reply:
I think the logic says,
If user is light or dark and their system doesn't support theme changes, we just use light/dark as that is what came from the local store. (and will be their only options)

If user is set to system or support for system themes, go get what is applied via the system.

Possibly this logic may be able to be simplified if this check was done outside of this component.

Reply:
Hmm.. no it doesn't. Just if it is set to system theme go get what is set or  it does not support system themes.. go get what is set? That does seem strange

Reply:
Based on my last reasoning/comment, I am not sure if we event need to check if it supportsSystemThemeChanges here...

Reply:
Meh.. I am confused.. it would seem the only time this check matters is if we have it set as system and system is not supported (How would that happen?).. cause then it would go get light or dark..

Reply:
Ah.. we default to system. :D Ok.. Definitely this logic would be less confusing if we did this `supportsSystemThemeChanges` up at the app store and set it to System or result of `getCurrentlyAppliedTheme` :) But... I think that could be a refactor for another PR

Reply:
Thanks for digging into this @tidy-dev!